crates/syntax/src/ast/make.rs

```diff
@@ -184,8 +184,11 @@ pub fn impl_trait(
     ty: ast::Path,
     ty_params: Option<ast::GenericParamList>,
 ) -> ast::Impl {
-    let ty_params = ty_params.map_or_else(String::new, |params| params.to_string());
-    ast_from_text(&format!("impl{ty_params} {trait_} for {ty}{ty_params} {{}}"))
+    // TODO : If this function is now correct we can also change `impl_` accordingly`
```

Comment:
`impl_` as the comment states should take `ty_args: Option<ast::GenericArgList>`, not `ty_params: Option<ast::GenericParamList>`. Likewise it would be better if we change `impl_trait` to the following signature:
```rs
pub fn impl_trait(
    trait_: ast::Path,
    ty: ast::Path,
    ty_params: Option<ast::GenericParamList>,
    ty_args: Option<ast::GenericArgList>,
) -> ast::Impl {
```